sync/upstream.py

```diff
@@ -425,14 +425,22 @@ def try_land_pr(self):
             else:
                 self.merge_sha = merge_sha
                 self.finish("wpt-merged")
-                # Delete the remote branch after a merge
-                self.git_wpt.remotes.origin.push(self.remote_branch, delete=True)
-                self.remote_branch = None
                 return True
         if msg is not None:
             logger.error(msg)
         return False
 
+    def finish(self, status="complete"):
+        super(UpstreamSync, self).finish(status)
+        if status in ("wpt-merged", "complete") and self.remote_branch:
+            # Delete the remote branch after a merge
+            try:
+                self.git_wpt.remotes.origin.push(self.remote_branch, delete=True)
+            except git.GitCommandError:
+                pass
```

Comment:
Worth a log entry? Remotes could add up if something is fundamentally broken. Or how are remaining remote branches tracked? At least those aren't getting reset to `None` here.